src/main/java/FIFOSchedular.java

```diff
@@ -0,0 +1,95 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class FIFOSchedular implements TaskSchedular {
+
+    List<Task> taskList;
+    Map<Integer, HashSet<UUID>> taskGroupsBasedOnPriority;
+    final int queueCapacity;
+
+    FIFOSchedular(int queueCapacity){
+        this.taskList = new ArrayList<Task>(queueCapacity);
+        this.queueCapacity = queueCapacity;
+        this.taskGroupsBasedOnPriority = new HashMap<Integer, HashSet<UUID>>();
+    }
+
+    public void addNewTaskToQueueAndMap(Task task) {
+
+        if( taskList.size() < queueCapacity){
+
+            taskList.add(task);
+
+            addTaskToTaskMapBasedOnPriority(task);
+
+            System.out.println("New task added to Task List.\n");
+            printTaskQueue();
+        }
+        else
+            System.out.println("Error : Can't add new task. Task List is full.\n");
+    }
+
+    public void printTaskQueue() {
+
+        System.out.println("Tasks currently present in Task List :\n");
+
+        for (Task task : taskList) {
+            System.out.println("Task Id :" + String.valueOf(task.getTaskId()) + ", Priority :" + String.valueOf(task.getTaskPriority())+"\n");
+        }
+    }
+
+    public Task fetchTaskFromQueueForExecution() {
+
+        if(!taskList.isEmpty()){
+            Task task = taskList.get(0);
+            taskList.remove(0);
```

Comment:
No we can't use Stack as it is a FIFO schedular. Queue is a possible candidate but if we use queue then design wise we also have to use Stack in LIFO schedular and Priority Queue in Priority based schedular. So to keep it simple I used List by adding queue behaviour. Same for other classes.